### ISSUE TITLE
feature: Add health-check (liveness and readiness) probes for Zebra

### ISSUE THREAD

**Issue by gustavovalverde** [ZcashFoundation/zebra#8830]
### Motivation

If we're deploying some/several Zebra nodes, and we need to continuously confirm if those nodes are running and behaving as expected, we should have a way to validate the nodes are **live** and **ready**, which would be defined as:
- Readiness:  Zebra is up, it's able to connect to other nodes, and respond to requests. If
- Liveness: The node is running, and it's synced and/or progressing with the sync. Ideally we should have states for this:
  - Ready to sync
  - Syncing
  - Synced

This would make nodes management easier, and more automatic. [Here is an explanation](https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/#define-a-liveness-http-request) on how this would work with Kubernetes, but this applies to most container orchestration tools.

### Specifications

Any solution should consider this endpoints will be publicly available and thus their security implications should be thoroughly evaluated.

Requirements and examples from other projects:
-  An implementation using a custom script (which I'd like to avoid): https://github.com/blockchain-etl/blockchain-kubernetes/blob/dev/charts/parity/templates/statefulset.yaml#L75-L80
- Similar requirement for Geth: https://github.com/golemfactory/concent-deployment/issues/44

### Complex Code or Requirements

- If using the RPC endpoints, then we should enable those in our nodes, and not allow public access to them, but just from our infrastructure. For example, we might be able to ping `getblockchaininfo` (sync progress) or `getinfo` (liveness) instead; and we could add more fields.
 - This wouldn't be a solution _for everyone_, but it would be a starting point

### Related Work

This was previously requested and partially done:
- https://github.com/ZcashFoundation/zebra/pull/479

**Comment by mpguerra** [comment]
@gustavovalverde did we do anything about this in the end?

**Comment by gustavovalverde** [comment]
@mpguerra no, but we should. Even more considering a few community members are running Kubernetes clusters with Zebra.

**Comment by emersonian** [comment]
We badly need this for our Kubernetes deployments, especially a "synced" readiness state for a zebrad to be included in our load balancers. Can this be prioritized before Zcashd is fully deprecated? Thank you.

I'd define two HTTP endpoints:
/healthy = Zebrad is running and connected to P2P network syncing
/ready = Zebrad is fully synced and ready for user traffic [ is_close_to_tip() , but I'd prefer no more than 2 blocks behind, configurable perhaps ]

User story:

As a Zcash node operator, I have 10 Zebrad processes/statefulsets running in Kubernetes. I only want to direct user traffic (lightwalletd) to the Zebrad processes that are fully synced.

The problem: When I scale up to 12 Zebrad instances, some of my user traffic will be directed to the Zebrad nodes that are not in sync, causing poor user experience when their wallets report a small block height and cannot sync if they were unlucky and connected to the new, syncing nodes.

**Comment by gustavovalverde** [comment]
Hey @emersonian thank you for the detailed use case, this helps to understand what to prioritize here. 

I recently created https://github.com/ZcashFoundation/zebra/pull/9886 as this was something I previously forgot to add. This is just to be able to do very basic liveness checks with Zebra. There's still some work to be done around this because when cookies are enabled the process lacks determinism and then it needs some workaround for the healtcheck to work.

I implemented this a few months ago, and after recent changes around Zebra and Zaino, now I need to update it to make it easier: https://github.com/ZcashFoundation/z3/blob/db63c82e76dde731cf36d7bf3d5feedfa03ac44e/docker-compose.yml#L29-L49 

I'll have a look at this next week
